source/common/formatter/substitution_formatter.cc

```diff
@@ -953,6 +953,16 @@ StreamInfoFormatter::StreamInfoFormatter(const std::string& field_name) {
           }
           return absl::nullopt;
         });
+  } else if (field_name == "VIRTUAL_CLUSTER_NAME") {
```

Comment:
wow this is a huge sequence of string comparisons. I'm assuming this is not in the request path. But even so maybe we could have a string->enum map and a switch statement?

not an issue for this PR but maybe leave a TODO/issue to address in a follow-up?

Reply:
Added a TODO at the top as a follow-up. I'll take it up next.